frontend/packages/console-app/src/components/nodes/NodesPage.tsx

```diff
@@ -40,60 +56,108 @@ const NodeTableHeader = () => {
       props: { className: tableColumnClasses[2] },
     },
     {
-      title: 'Machine',
-      sortField: "metadata.annotations['machine.openshift.io/machine']",
+      title: 'Pods',
+      sortField: 'status.capacity.pods',
       transforms: [sortable],
       props: { className: tableColumnClasses[3] },
     },
     {
-      title: '',
+      title: 'Memory',
+      sortFunc: 'nodeMemory',
+      transforms: [sortable],
       props: { className: tableColumnClasses[4] },
     },
+    {
+      title: 'CPU',
+      sortFunc: 'nodeCPU',
+      transforms: [sortable],
+      props: { className: tableColumnClasses[5] },
+    },
+    {
+      title: 'Filesystem',
+      sortFunc: 'nodeFS',
+      transforms: [sortable],
+      props: { className: tableColumnClasses[6] },
+    },
+    {
+      title: 'Created At',
+      sortField: 'metadata.creationTimestamp',
+      transforms: [sortable],
+      props: { className: tableColumnClasses[7] },
+    },
+    {
+      title: '',
+      props: { className: tableColumnClasses[8] },
+    },
   ];
 };
 NodeTableHeader.displayName = 'NodeTableHeader';
 
-const NodesTableRow: React.FC<NodesTableRowProps> = ({ obj: node, index, key, style }) => {
-  const { name: machineName, namespace: machineNamespace } = getNodeMachineNameAndNamespace(node);
-  const nodeName = getName(node);
-  const nodeUID = getUID(node);
-  return (
-    <TableRow id={nodeUID} index={index} trKey={key} style={style}>
-      <TableData className={tableColumnClasses[0]}>
-        <ResourceLink kind={referenceForModel(NodeModel)} name={nodeName} title={nodeUID} />
-      </TableData>
-      <TableData className={tableColumnClasses[1]}>
-        <NodeStatus node={node} />
-      </TableData>
-      <TableData className={tableColumnClasses[2]}>
-        <NodeRoles node={node} />
-      </TableData>
-      <TableData className={tableColumnClasses[3]}>
-        {machineName && (
-          <ResourceLink
-            kind={referenceForModel(MachineModel)}
-            name={machineName}
-            namespace={machineNamespace}
-          />
-        )}
-      </TableData>
-      <TableData className={tableColumnClasses[4]}>
-        <ResourceKebab actions={menuActions} kind={referenceForModel(NodeModel)} resource={node} />
-      </TableData>
-    </TableRow>
-  );
+const mapStateToProps = ({ UI }) => ({
+  metrics: UI.getIn(['metrics', 'node']),
+});
+
+type NodesRowMapFromStateProps = {
+  metrics: NodeMetrics;
 };
+
+const NodesTableRow = connect<NodesRowMapFromStateProps, null, NodesTableRowProps>(mapStateToProps)(
+  ({ obj: node, index, key, style, metrics }: NodesTableRowProps & NodesRowMapFromStateProps) => {
+    const nodeName = getName(node);
+    const nodeUID = getUID(node);
+    const freeMem: number = Number(_.get(metrics, ['freeMemory', nodeName]));
```

Comment:
Don't use `_.get`. Prefer optional chaining, which is type safe. (Here and below.)

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#optional-chaining

We need to handle `NaN` and missing metrics.

`: number` isn't a problem, but it will be inferred by TypeScript and is unnecessary. (Same for the types just below.)

Reply:
ack. Will use a different approach.